### PR TITLE
#164798171 Pagination support for Articles

### PR DESCRIPTION
**Pull request by dorothyas**
**what does this PR do?**
- this enables users to view a specified number of articles and go to the next page for more articles

**How should this be manually tested?**

- Fetch this branch git fetch origin ft-pagination-support-articles-164798171
- Checkout to the branch git checkout ft-pagination-support-articles-164798171
- Run npm install
- Run npm start to view the landing page 
- Login to the app on /Login
- Check out the paginated articles on the dashboard, and click next to view other articles.

**What are the relevant Pivotal Tracker stories?**

[#164798171](https://www.pivotaltracker.com/n/projects/2320796/stories/164798171)

**Screenshots**

- The First Articles on the landing page

Screenshot 2019-05-20 at 16 43 25

- The Next Articles when "next" is clicked.

Screenshot 2019-05-20 at 16 40 37

### CODE REVIEW

File: src/actions/profileActions.js [hunk 1]
```diff
@@ -38,6 +38,7 @@ export const saveImageActions = (url, username) => (dispatch) => {
     .put(`${process.env.baseURL}/profiles/${username}`, data)
     .then(() => {
       localStorage.setItem('image', url);
+      // eslint-disable-next-line no-restricted-globals
```
Comment (ezrogha) [inline src/actions/profileActions.js]: Hey @dorothyas, 
Kindly get rid of this comment

File: src/components/NavBarAccountSection/index.jsx [hunk 2]
```diff
@@ -2,6 +2,7 @@ import React from 'react';
 
 import './NavBarAccountSection.scss';
 
+// eslint-disable-next-line react/prop-types
```
Comment (ezrogha) [inline src/components/NavBarAccountSection/index.jsx]: Please remove this comment

File: src/pages/Dashboard/index.jsx [hunk 3]
```diff
@@ -4,33 +4,78 @@ import PropTypes from 'prop-types';
...
+
+  fetchPrevious = (event) => {
+    event.preventDefault();
+    const { fetchNext } = this.props;
+    fetchNext(localStorage.previous);
+  };
+
   render() {
-    const { articles } = this.props;
+    const { articles, firstarticles, paginatearticles } = this.props;
```
Comment (ezrogha) [inline src/pages/Dashboard/index.jsx]: Be consistent with the use of camelcase

File: src/pages/Landing/index.jsx [hunk 4]
```diff
@@ -1,20 +1,38 @@
+/* eslint-disable react/prop-types */
+/* eslint-disable react/destructuring-assignment */
```
Comment (ezrogha) [inline src/pages/Landing/index.jsx]: Please remove these comments

**Review by ezrogha — CHANGES_REQUESTED** [review]
Hey @dorothyas,
Good work here,
kindly work on this feedback

File: src/pages/Landing/index.jsx [hunk 5]
```diff
@@ -30,26 +48,54 @@ export class Landing extends Component {
             <a href="/login">Account</a>
           </div>
         </div>
-        <ArticleContainer articles={articles} />
-        <br />
-        <br />
-        <br />
+        <div>
+          <div>
+            {paginatearticles.length === 0 ? <ArticleContainer articles={firstarticles} /> : <ArticleContainer articles={paginatearticles} />}
```
Comment (felixkiryowa) [inline src/pages/Landing/index.jsx:48]: @dorothyas Formatting of this line is required

**Review by felixkiryowa — DISMISSED** [review]

**Review by ezrogha — DISMISSED** [review]
Great work with the feedback @dorothyas,
I realise the coverage has slightly dropped. 
please work on this

**Review by hadijahkyampeire — APPROVED** [review]
Well done but we might need a follow up PR for this after the backend fix

**Pull request merged**